Client/mods/deathmatch/logic/luadefs/CLuaVehicleDefs.cpp

```diff
@@ -240,6 +241,7 @@ void CLuaVehicleDefs::AddClass(lua_State* luaVM)
     lua_classfunction(luaVM, "getWheelScale", "getVehicleWheelScale");
     lua_classfunction(luaVM, "getModelWheelSize", "getVehicleModelWheelSize");
     lua_classfunction(luaVM, "getWheelFrictionState", "getVehicleWheelFrictionState");
+    lua_classfunction(luaVM, "getEntryPoints", ArgumentParser<OOP_GetVehicleEntryPoints>);
```

Comment:
```cpp
lua_classfunction(luaVM, "getEntryPoints", "getVehicleEntryPoints");
```

Reply:
I've tested this. Does not work with an array of vectors. Changing the argument parser to make this work is out of the scope of this PR.

Reply:
> I've tested this. Does not work with an array of vectors. Changing the argument parser to make this work is out of the scope of this PR.

It works fine for me for both methods
![image](https://github.com/multitheftauto/mtasa-blue/assets/43095317/d3fbc88b-88c1-42be-9d19-c3c67fbb6866)
![image](https://github.com/multitheftauto/mtasa-blue/assets/43095317/595f5e16-3169-4e99-ad25-4592f95bada7)

Reply:
Array of vectors works fine for me. Do you have any compilation issues?

Reply:
No compilation issues, however non-OOP functions aren't supposed to return vectors at all. Those are OOP only.

Reply:
If `CVector` is too slow then `std::array` seems like an obvious choice here

Reply:
It changes what is output in Lua, that's the point. OOP functions returns Vector3 objects for positions and non-OOP functions don't

Reply:
> It changes what is output in Lua, that's the point. OOP functions returns Vector3 objects for positions and non-OOP functions don't

Thats what I mean. Either have them both return CVector or use `std::array`. Splitting them up into 2 dont make much sense especially when you'd want to get individual entry point locations.

Reply:
They are both returning std:array though, the difference is that the OOP function gives us a table with 4 Vector3 objects, one for each entrypoint. The non-OOP version gives us a table that by itself has 4 tables (one for each entrypoint) that each contain 3 floats (the x, y & z positions).

If you can find a way to combine these functions, yet have those intended outputs, let me know. I can't find a way to do it.

Reply:
> That's not what a POP function is supposed to do in MTA

What is POP function supposed to do?

> Similarly changing the OOP one to std:array with floats will make it so that it's not returning those Vector3 objects, which a OOP function is expected to do.

Why is it expected to do so?

Reply:
@MegadreamsBE is right, the method should be split (as it already is) so it can return the correct value between POP/OOP.

Reply:
> > That's not what a POP function is supposed to do in MTA
> 
> What is POP function supposed to do?

Vector3 (which CVector will result into thanks to the argument parser) is a OOP-only object. It should never be returned by POP functions. It's also why you don't see it being used outside of OOP functions.

> > Similarly changing the OOP one to std:array with floats will make it so that it's not returning those Vector3 objects, which a OOP function is expected to do.
> 
> Why is it expected to do so?

Because that's how the OOP functions in MTA work. It's an expectation for them to return Vector3 objects that can easily be manipulated using the methods defined here: https://wiki.multitheftauto.com/wiki/Vector/Vector3.

I'm merely keeping my functions consistent with the remainder of the code base.

Reply:
If all OOP functions return Vector then its fine

Reply:
> If all OOP functions return Vector then its fine, nevermind my review then

Jup, that's consistently the case.